Remotion/ObjectBinding/Web/UI/Controls/BocTreeNode.cs

```diff
@@ -239,17 +254,20 @@ protected override string DisplayedTypeName
     get { return "PropertyNode"; }
   }
 
+  [MemberNotNull (nameof (_property))]
   private void EnsureProperty()
   {
     if (_property != null)
       return;
 
-    BusinessObjectTreeNode parentNode = (BusinessObjectTreeNode) ParentNode;
+    BusinessObjectTreeNode? parentNode = (BusinessObjectTreeNode?) ParentNode;
     if (parentNode == null)
       throw new InvalidOperationException ("BusinessObjectPropertyTreeNode with ItemID '" + ItemID + "' has no parent node but property nodes cannot be used as root nodes.");
 
-    IBusinessObjectProperty property = parentNode.BusinessObject.BusinessObjectClass.GetPropertyDefinition (ItemID);
-    Property = (IBusinessObjectReferenceProperty) property;
+    Assertion.IsNotNull (parentNode.BusinessObject, "parentNode.BusinessObject must not be null.");
+
+    IBusinessObjectProperty? property = parentNode.BusinessObject.BusinessObjectClass.GetPropertyDefinition (ItemID);
+    _property = (IBusinessObjectReferenceProperty) Assertion.IsNotNull (property, "No property definition exists for the id '{0}'.", ItemID);
```

Comment:
`"Property '{0}' does not exist on class '{1}'.", ItemID, parentNode.BusinessObject.BusinessObjectClass.Identifer);